### PR TITLE
Roundtrip test helper for external types

### LINKED ISSUE #54698: need deep copy fuzz test helper
Writing an API repo, you need to check that your deep copy methods work in isolation.  We should provide a test function in apimachinery for this purpose outside of conversions.

### PR DESCRIPTION
**Pull request by soltysh**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #54698

**Special notes for your reviewer**:
/assign @deads2k @sttts 
Since you've asked for it in the original issue. The FIXME I've pointed out to is worth addressing, but I'd prefer to do it afterwards. Do we want to implement appropriate test for all the api groups from `k8s.io/api` ?

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by sttts** [comment]
gofmt

/approve

**Comment by soltysh** [comment]
Updated, @sttts ptal

File: staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go [hunk 1]
```diff
@@ -138,6 +138,26 @@ func roundTripTypes(t *testing.T, scheme *runtime.Scheme, codecFactory runtimese
 	}
 }
 
+func RoundTripExternalTypes(t *testing.T, scheme *runtime.Scheme, codecFactory runtimeserializer.CodecFactory, fuzzer *fuzz.Fuzzer, nonRoundTrippableTypes map[schema.GroupVersionKind]bool) {
+	roundTripExternalTypes(t, scheme, codecFactory, fuzzer, nonRoundTrippableTypes)
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go]: why this indirection?

File: staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go [hunk 2]
```diff
@@ -138,6 +138,26 @@ func roundTripTypes(t *testing.T, scheme *runtime.Scheme, codecFactory runtimese
 	}
 }
 
+func RoundTripExternalTypes(t *testing.T, scheme *runtime.Scheme, codecFactory runtimeserializer.CodecFactory, fuzzer *fuzz.Fuzzer, nonRoundTrippableTypes map[schema.GroupVersionKind]bool) {
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go]: godoc

File: staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go (hunk 1 above)
Comment (soltysh) [reply staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go]: Copy&paste - removed.

**Comment by soltysh** [comment]
@sttts update, ptal

File: staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go [hunk 3]
```diff
@@ -138,6 +138,24 @@ func roundTripTypes(t *testing.T, scheme *runtime.Scheme, codecFactory runtimese
 	}
 }
 
+// RoundTripExternalTypes applies the round-trip test to all external round-trippable Kinds
+// in the scheme.  It will skip all the GroupVersionKinds in the nonRoundTripExternalTypes list .
+func RoundTripExternalTypes(t *testing.T, scheme *runtime.Scheme, codecFactory runtimeserializer.CodecFactory, fuzzer *fuzz.Fuzzer, nonRoundTrippableTypes map[schema.GroupVersionKind]bool) {
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go]: lgtm. Can we wire this somewhere?

**Comment by soltysh** [comment]
@sttts the 2nd commit adds the test verifying the method is working as expected and that the external k8s api is passing `DeepCopy` tests.

File: staging/src/k8s.io/api/pkg/testing/deepcopy_test.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (sttts) [inline staging/src/k8s.io/api/pkg/testing/deepcopy_test.go]: how about putting it into the root package as roundtrip_test.go?

Comment (soltysh) [reply staging/src/k8s.io/api/pkg/testing/deepcopy_test.go]: Yeah, I wasn't sure where to put it, root package sgtm.

**Comment by soltysh** [comment]
@sttts updated, ptal

File: staging/src/k8s.io/api/deepcopy_test.go [hunk 5]
```diff
@@ -0,0 +1,148 @@
+/*
```
Comment (sttts) [inline staging/src/k8s.io/api/deepcopy_test.go]: deepcopy is only one part. Why not roundtrip_test.go? We have that already in the internal type packages.

**Comment by soltysh** [comment]
@sttts updated, ptal

**Comment by soltysh** [comment]
/assign @smarterclayton 
for approval

**Pull request merged**